common/scala/src/main/scala/whisk/core/entity/InstanceId.scala

```diff
@@ -30,7 +30,8 @@ import whisk.core.entity.ControllerInstanceId.MAX_NAME_LENGTH
  */
 case class InvokerInstanceId(val instance: Int,
                              uniqueName: Option[String] = None,
-                             displayedName: Option[String] = None) {
+                             displayedName: Option[String] = None,
+                             val userMemory: Int = 1024) {
```

Comment:
* ByteSize(“....”) needs more space than Int.
* If i don't give the default value:1024, other relative build InvokerInstanceId Object, needs to modify the codes. And i keep the default value(1024), the same as {{ user.memory }} maybe good.

Reply:
For test cases, i  just modified original test cases to support `different amount of userMemory`

Reply:
Keeping defaults in multiple places will eventually rot I believe.

Regarding ByteSize needing more space: It sounds like that's premature optimization and a concern of serdes, we should strive to keep types in the code as self-explanatory as possible, thus I'd also use ByteSize here.

Reply:
I modified `ByteSize` in local, when build, reported below error
```
common/scala/src/main/scala/whisk/core/entity/InstanceId.scala:48: could not find implicit value for evidence parameter of type whisk.core.entity.InvokerInstanceId.JF[whisk.core.entity.ByteSize]
  implicit val serdes = jsonFormat4(InvokerInstanceId.apply)
```
I tried to add below `import` in `/scala/src/main/scala/whisk/core/entity/InstanceId.scala` , but the problem still exist
```
import whisk.core.entity.size._
```
@markusthoemmes , do you have any idea about this problem?

Reply:
@markusthoemmes ,already modified.
Another thing, i want to share.

Why i changed `whisk.core.entity.Size.scala`'s `val size = matcher.group(1).toInt` to `toLong`,
because if use `toInt` and if the pingMessage is like below 
```
{"name":{"instance":2,"uniqueName":"2","userMemory":"10737418240 B"}}
```
When deserialize，will report below error
```
failed processing message: {"name":{"instance":2,"uniqueName":"2","userMemory":"10737418240 B"}} with java.lang.NumberFormatException: For input string: "10737418240"
```
Because `10737418240`  is very big for `Int`, so it is necessary to change to `toLong` here.